packages/react-dialog/Spec.md

```diff
@@ -2,62 +2,303 @@
+    </DialogFooter>
+  </Dialog>
+
+```
+
+#### v0
+
+In v0, the Dialog component expects all the content through props, including the content, actions etc. The dialog component uses the `trigger` prop that expects a React component to control its visibility. The element passed to this prop will be rendered in-place where the dialog is defined.
+
+[Documentation for v0 Dialog](https://fluentsite.z22.web.core.windows.net/components/dialog/definition)
+
+```jsx
+<Dialog
+  cancelButton="Connect protocol"
+  confirmButton="Transmit capacitor"
+  content="Connect driver"
+  header="Transmit capacitor"
+  headerAction="Generate protocol"
+  trigger={<Button content="A trigger" />}
+/>
+```
+
+## API proposal
+
+Dialog component is composed by 3 sub-components: Header, Body and Footer.
+
+### Components
+
+| Component    | Purpose                                                  |
+| ------------ | -------------------------------------------------------- |
+| Dialog       | The main wrapper component.                              |
+| DialogHeader | (optional) Component for the title and the close button. |
+| DialogBody   | (optional) Component for the main content of dialog.     |
+| DialogFooter | (optional) Component for the main actions of dialog.     |
+
+## Dialog
+
+The dialog is a container which handles styling (border, background etc.) and the overlay (dimmed background).
+
+### Anatomy
+
+![Visual anatomy of Dialog component](./assets/dialog-anatomy.png)
+
+### API
+
+| Property | Values                        | Default   | Purpose                                  |
+| -------- | ----------------------------- | --------- | ---------------------------------------- |
+| type     | `modal`, `non-modal`, `alert` | `modal`   | Dialog variations                        |
+| open     | boolean                       | `false`   | Set to `true` when the dialog is visible |
+| overlay  | _slot_                        | undefined | Dimmed background of dialog              |
+
+> ⚠️ _Pending issue: `draggable`, the dragging functionality of the dialog is under consideration. This might be excluded from the dialog implementation._
+
+- `type` property (dialog variations):
+
+  - `modal`: When this type of dialog is open, the rest of the page is dimmed out and cannot be interacted with. The tab sequence is kept within the dialog and moving the focus outside the dialog will imply closing it. This is the default type of the component.
+
+  - `non-modal`: When a non-modal dialog is open, the rest of the page is not dimmed out and users can interact with the rest of the page. This also implies that the tab focus can move outside the dialog when it reaches the last focusable element.
+
+  - `alert`: is a special type of modal dialogs that interrupts the user's workflow to communicate an important message or ask for a decision. These dialogs are not dismissable, neither by escape key or by clicking outside the dialog.
```

Comment:
so how are `alert` dialogs removed from the page if they cannot be dismissed ?

Reply:
oh I see that actions inside the dialog can still dismiss it, can you update this description to reflect that ?

Reply:
yes the actions inside the dialog should be the only ones that can dismiss the dialog. This is like a special kind of dialog (it is also treated differently by the screen readers). I will add this to the description, thanks for pointing that out.